### ISSUE TITLE
Interval for firing button listeners when holding down enter key

### ISSUE THREAD

**Issue by jessegreenberg** [phetsims/sun#424]
This came from https://github.com/phetsims/resistance-in-a-wire/issues/193. If using a keyboard PhET buttons can be clicked as fast as the user presses them. This includes pressing and holding enter on a button, in which case the button is clicked every couple of milliseconds, and that delay is determined by OS settings.

For example, got to https://phet-dev.colorado.edu/html/resistance-in-a-wire/1.6.0-rc.2/phet/resistance-in-a-wire_en_phet.html and tab to the Reset All button then press and hold enter on it. You will her a stream of the "Reset All" sounds as the button is fired continuously.

Some button models in sun already have "fireOnHold" configurations and delays hopefully a11y can use these directly.

**Comment by emily-phet** [comment]
@jessegreenberg @zepumph Let's wait on this fix for the next round of sims and move forward with RIAW and Friction.

**Comment by jessegreenberg** [comment]
Thanks @emily-phet sounds good, in that case removing the blocks-publication label.

**Comment by jessegreenberg** [comment]
A "button specific fix" was added to the Reset All button for RIAW, and we should do this for the sound toggle button for the upcoming friction release.

**Comment by zepumph** [comment]
I may not understand this issue fully, so please pardon me if this is not right. What about the keyboard help button, could someone spam both the nav bar button and the dialog's "x" to close it out really fast and encounter this issue too? This seems so sloppy to require this on each instance. Is there any hope of having a general solution for this?

**Comment by jessegreenberg** [comment]
The original report of this issue was that pressing and holding enter on the Reset All button caused VO to announce the reset alert indefinitely, and the same will happen for alerts from the sound toggle button. The case you mention is definitely related to this but not problematic for a11y in the same way because it doesn't involve alerts.

I feel confident that we will have a general solution for this, but I don't feel confident that this will be completed by the end of the week or that the changes can be easily patched into the release branch of RIAW without starting fresh with a new round of testing.

Totally agree it is sloppy to have the workaround at two instances, but this is not the long term solution.

**Comment by zepumph** [comment]
Understood thanks!

**Comment by jessegreenberg** [comment]
Adding back the blocks-publication label as this should be fixed before the round of a11y sims is published. This does not apply to Friction and RIAW.

**Comment by pixelzoom** [comment]
This and https://github.com/phetsims/sun/issues/463 are related.  Sun buttons have a fireOnHold option that is currently not integrated with a11y.  This feature should work the same regardless of whether the user is holding down a pointer or a key.

**Comment by pixelzoom** [comment]
The title of this issue is "Interval for firing button listeners when holding down enter key", but in https://github.com/phetsims/scenery/issues/939#issue-404025451 we concluded that Enter and Space should behave the same.   So are we planning to support fire-on-hold for Space?  And how do we do that when we only get `click` on release when using 'button' role?

**Comment by jessegreenberg** [comment]
> Sun buttons have a fireOnHold option that is currently not integrated with a11y. This feature should work the same regardless of whether the user is holding down a pointer or a key.

In https://github.com/phetsims/scenery/issues/939 we concluded that for screen reader and a11y interaction buttons will continue to operate in a way that is native for the web. So buttons will generally fire continuously when pressing and holding down enter and fire once on release of `spacebar`.

For unique buttons that require specific intervals when holding down enter, or that *must* only fire on key release we are going to proceed with https://github.com/phetsims/scenery/issues/939#issuecomment-480620125, which involves changing the markup for certain buttons and `keydown` and `keyup` listeners to PressListener so that the following PushButtonModel options will work in a way that matches pointer input.

```js
      fireOnDown: false, // true: fire on pointer down; false: fire on pointer up if pointer is over button
      fireOnHold: false, // is the fire-on-hold feature enabled?
      fireOnHoldDelay: 400, // start to fire continuously after pressing for this long (milliseconds)
      fireOnHoldInterval: 100, // fire continuously at this interval (milliseconds), same default as in ButtonModel
```
That work is going to be done in https://github.com/phetsims/scenery/issues/939 so this issue can be closed.  Please see https://github.com/phetsims/scenery/issues/939 for discussion about why PhET decided not to do a project-specific solution for accessible buttons.
